packages/playwright-crawler/src/internals/adaptive-playwright-crawler.ts

```diff
@@ -186,11 +186,17 @@ export interface AdaptivePlaywrightCrawlerOptions
 
     /**
      * An optional callback used in rendering type detection. On each detection, the result of the plain HTTP run is compared to that of the browser one.
-     * If the callback returns true, the results are considered equal and the target site is considered static.
+     * If a callback is provided, the contract is as follows:
+     *   It the callback returns true or 'equal', the results are considered equal and the target site is considered static.
+     *   If it returns false or 'different', the target site is considered client-rendered.
+     *   If it returns 'inconclusive', the detection result won't be used.
      * If no result comparator is specified, but there is a `resultChecker`, any site where the `resultChecker` returns true is considered static.
      * If neither `resultComparator` nor `resultChecker` are specified, a deep comparison of returned dataset items is used as a default.
      */
-    resultComparator?: (resultA: RequestHandlerResult, resultB: RequestHandlerResult) => boolean;
+    resultComparator?: (
+        resultA: RequestHandlerResult,
+        resultB: RequestHandlerResult,
+    ) => boolean | 'equal' | 'different' | 'inconclusive';
```

Comment:
křížek after funus, but could this be an enum? Possibly something like this:

![image](https://github.com/user-attachments/assets/0fd5aede-48e4-43be-b1fe-917b2412229f)

So the `equal` value is truthy, and the `different` is falsy, so the condition can be written like this:

![image](https://github.com/user-attachments/assets/686c8541-113e-4792-ad39-1e5ea38f01a0)3

It's rather a nit, but could make the DX better (and our code cleaner).